content/docs/self-hosted-runners.md

```diff
@@ -207,25 +245,52 @@ Click below to see credentials needed for supported cloud service providers.
 
 </details>
 
-## Using on-premise machines as self-hosted runners
+<details>
+
+#### GCP
+
+Either one of:
+
+- `GOOGLE_APPLICATION_CREDENTIALS_DATA`: the **contents** of a service account
+  JSON file, or
+- `GOOGLE_APPLICATION_CREDENTIALS`: the **path** to the JSON file.
 
-You can also use `cml-runner` to set up a self-hosted runner on your local
-machine or on-premise GPU cluster.
+The former is more convenient for CI/CD scenarios, where secrets are (usually)
+provisioned through environment variables instead of files.
 
-```dvc
+</details>
+
+<details>
+
+#### Kubernetes
+
+- `KUBERNETES_CONFIGURATION`: the **contents** of a `kubeconfig` file
```

Comment:
```suggestion
- `KUBERNETES_CONFIGURATION` with the **contents** of a `kubeconfig` file
```
Same as above: https://github.com/iterative/cml.dev/pull/105/files#r708579464

Reply:
Also, [`kubeconfig`](https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/) probably needs a link to https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig for novice users.

Reply:
yes all the providers need to be fleshed out later with proper tutorials